### DISCUSSION TITLE
Inpainting seems to clear original image

### DISCUSSION THREAD

**Discussion by viraptor** [Comfy-Org/ComfyUI#2274] [Q&A]
I've got the following workflow which I hoped would result in inpainting of one image into an area of another.
But the result is a mess:

<img width="1316" alt="Screenshot 2023-12-13 at 1 28 26 am" src="https://github.com/comfyanonymous/ComfyUI/assets/188063/d7af1fec-ecb2-4166-b170-adad7113c20e">

What am I missing here? Using the SD 1.5 pruned checkpoint from https://huggingface.co/runwayml/stable-diffusion-v1-5
(I was after img2img like behaviour here)

Joining the conditioning does the opposite, but the inpainted stuff is missing instead.

<img width="1324" alt="Screenshot 2023-12-13 at 1 24 21 am" src="https://github.com/comfyanonymous/ComfyUI/assets/188063/e8f885f0-2355-4d14-ac01-271074b755b8">

Trying to use an inpainting model fails similarly:

<img width="1313" alt="Screenshot 2023-12-13 at 1 33 40 am" src="https://github.com/comfyanonymous/ComfyUI/assets/188063/a280b812-de75-4ea4-ae18-b9c15a773173">

**Comment by jn-jairo** [comment]
Ok the trick for inpainting is:

---

### For standard models (not inpaint model)

Use "Set Latent Noise Mask" to set the mask and denoise cannot be 1 or it will be totally different, something like 0.6-0.8 is good to have enough noise to make the changes, you can go lower but the lower you get the harder it will be to have changes, it is also good to blur the mask to avoid seams, there is no node to blur mask but you can convert the mask to image to blur it and convert back to mask.

### For inpaint models

Use "VAE Decode (for Inpainting)" to set the mask and the denoise must be 1, inpaint models only accept denoise 1, anything else will result in a trash image.

---

The conditioning set mask is not for inpaint workflows, if you want to generate images with objects in a specific location based on the conditioning you can see the examples in [here](https://comfyanonymous.github.io/ComfyUI_examples/area_composition/).

Another thing, the official stable diffusion model is not good for inpainting, I suggest you get a custom model, for realistic pictures I like to use the [Realistic Vision](https://civitai.com/models/4201) it is good for inpaint even with the version not made for inpaint, but that is just my opinion, you can use any model you want, just find a more recent model, the official ones are not the top quality models at all.
